src/Microsoft.PowerShell.Commands.Management/commands/management/NewPropertyCommand.cs

```diff
@@ -232,23 +232,12 @@ public IEnumerable<CompletionResult> CompleteArgument(
                 yield break;
             }
 
-            string quote = CompletionCompleters.HandleDoubleAndSingleQuote(ref wordToComplete);
-            var propertyTypePattern = WildcardPattern.Get(wordToComplete + "*", WildcardOptions.IgnoreCase);
-
-            foreach (string propertyType in s_RegistryPropertyTypes)
+            foreach (CompletionResult result in CompletionCompleters.EnumerateQuotedAndUnquotedCompletionText(
+                wordToComplete,
+                possibleCompletionValues: s_RegistryPropertyTypes,
+                toolTipMapping: GetRegistryPropertyTypeToolTip))
             {
-                if (propertyTypePattern.IsMatch(propertyType))
-                {
-                    string completionText = quote == string.Empty
-                        ? propertyType
-                        : quote + propertyType + quote;
-
-                    yield return new CompletionResult(
-                        completionText,
-                        propertyType,
-                        CompletionResultType.ParameterValue,
-                        GetRegistryPropertyTypeToolTip(propertyType));
-                }
+                yield return result;
```

Comment:
It seems we can remove rest yield-s too. I'm not against using yield-s, but still, the code is better readable and better generated without them.
```c#
            => IsRegistryProvider(fakeBoundParameters) ?
                    CompletionCompleters.EnumerateQuotedAndUnquotedCompletionText(
                        wordToComplete,
                        possibleCompletionValues: s_RegistryPropertyTypes,
                        toolTipMapping: GetRegistryPropertyTypeToolTip))
                  : Array.Empty<CompletionResult>;
            }

```

Reply:
Yep this is nicer, I have refactored to expression body.